pkgs/tools/package-management/ciel/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, systemd
+, dbus
+, openssl
+, libssh2
+, fetchurl
+, xz
+, pkg-config
+, libgit2
+, zlib
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ciel";
+  version = "3.1.3";
+
+  src = fetchFromGitHub {
+    owner = "AOSC-Dev";
+    repo = "ciel-rs";
+    rev = "02eac316765f27239b7b17176df812da702895cd";
+    hash = "sha256-WzrOCiIOdg3fBLKNjCPlr/XGrXC2424hO1nrwWXjx2A=";
+  };
+
+  nativeBuildInputs = [ pkg-config zlib ];
+
+  doCheck = false;
+
+  buildInputs = [ systemd dbus openssl libssh2 libgit2 xz ];
+
+  cargoLock = {
+    lockFile = fetchurl {
```

Comment:
This introduces IFD (which is not allowed in nixpkgs). Actually we ARE vendoring lock files into nixpkgs now, see: https://github.com/NixOS/nixpkgs/pull/217084